plot_app/configured_plots.py

```diff
@@ -20,8 +22,168 @@
+        plots.append(widgetbox(div, width=int(plot_width*0.9)))
+
+    for index, axis in enumerate(['roll', 'pitch', 'yaw']):
+        axis_name = axis.capitalize()
+        # rate
+        data_plot = DataPlot(data, plot_config, 'actuator_controls_0',
+                             y_axis_label='[deg/s]', title=axis_name+' Angular Rate',
+                             plot_height='small',
+                             x_range=x_range)
+
+        thrust_max = 200
+        actuator_controls = data_plot.dataset
+        time_controls = actuator_controls.data['timestamp']
+        thrust = actuator_controls.data['control[3]'] * thrust_max
+        # downsample if necessary
+        max_num_data_points = 4.0*plot_config['plot_width']
+        if len(time_controls) > max_num_data_points:
+            step_size = int(len(time_controls) / max_num_data_points)
+            time_controls = time_controls[::step_size]
+            thrust = thrust[::step_size]
+        if len(time_controls) > 0:
+            # make sure the polygon reaches down to 0
+            thrust = np.insert(thrust, [0, len(thrust)], [0, 0])
+            time_controls = np.insert(time_controls, [0, len(time_controls)],
+                                      [time_controls[0], time_controls[-1]])
+
+        p = data_plot.bokeh_plot
+        p.patch(time_controls, thrust, line_width=0, fill_color='#555555',
+                fill_alpha=0.4, alpha=0, legend='Thrust [0, {:}]'.format(thrust_max))
+
+        data_plot.change_dataset('vehicle_attitude')
+        data_plot.add_graph([lambda data: (axis+'speed', np.rad2deg(data[axis+'speed']))],
+                            colors3[0:1], [axis_name+' Rate Estimated'], mark_nan=True)
+        data_plot.change_dataset('vehicle_rates_setpoint')
+        data_plot.add_graph([lambda data: (axis, np.rad2deg(data[axis]))],
+                            colors3[1:2], [axis_name+' Rate Setpoint'],
+                            mark_nan=True, use_step_lines=True)
+        axis_letter = axis[0].upper()
+        rate_int_limit = '(*100)'
+        # this param is MC/VTOL only (it will not exist on FW)
+        rate_int_limit_param = 'MC_' + axis_letter + 'R_INT_LIM'
+        if rate_int_limit_param in ulog.initial_parameters:
+            rate_int_limit = '[-{0:.0f}, {0:.0f}]'.format(
+                ulog.initial_parameters[rate_int_limit_param]*100)
+        data_plot.change_dataset('rate_ctrl_status')
+        data_plot.add_graph([lambda data: (axis, data[axis+'speed_integ']*100)],
+                            colors3[2:3], [axis_name+' Rate Integral '+rate_int_limit])
+        plot_flight_modes_background(data_plot, flight_mode_changes)
+
+        if data_plot.finalize() is not None: plots.append(data_plot.bokeh_plot)
+
+        # PID response
+        if not pid_analysis_error:
+            try:
+                gyro_rate = np.rad2deg(sensor_combined.data['gyro_rad['+str(index)+']'])
```

Comment:
Should probably use the rates from `rate_ctrl_status` instead of `sensor_combined`. https://github.com/PX4/Firmware/blob/master/src/modules/mc_att_control/mc_att_control_main.cpp#L755-L767

Reply:
The signals are pretty much the same, when I checked a while back, but you're right we should use that one. Since it's also logged at full rate we can do that. I'll look into it.